storage/product_storage.py

```diff
@@ -1,8 +1,66 @@
-from model.product_model import Product
+import logging
+from model.product import Product
+from .db_conn import db_connection
 
-class ProductStorage:   
+class ProductStorage: 
     def __init__(self):
-        self.products = []
+        self.logger = logging.getLogger(__name__)
+        self.db = db_connection
+    
+    def get_product_by_id(self, id: str) -> Product:
+        self.logger.info("Getting product in DB")
+        try:
+            cursor = self.db.cursor()
+            sql_query = """
+                SELECT id, name, description, price, quantity, active, created_at, updated_at
+                FROM products
+                WHERE id = %s;
+            """
+        
+            cursor.execute(sql_query, (id,))
+            result = cursor.fetchone()
+            if result:
+                product = Product(
+                    id = result[0],
+                    name = result[1],
+                    description = result[2],
+                    price = result[3],
+                    quantity = result[4],
+                    active = result[5],
+                    created_at = result[6],
+                    updated_at = result[7]
+                )
+                return product
+            else: 
+                raise Exception(f"Product not found with id {id}")
+        except Exception as ex: 
+            self.logger.error(f"Failed to get product by id={id} in DB. Error: {ex}")
+            raise ex
+        finally: 
+            cursor.close()
```

Comment:
Experimente usar um gerenciador de contexto para controlar a abertura e fechamento do cursor 
https://www.geeksforgeeks.org/with-statement-in-python/